docs/semgrep-ci/manual-ci-setup.md

```diff
@@ -0,0 +1,516 @@
+5. Customize the job's behavior, such as its rules to scan and files to ignore.
+
+#### CircleCI code snippet
+
+```yaml
+version: 2.1
+jobs:
+  semgrep-scan:
+    parameters:
+      default_branch:
+        type: string
+        default: main
+    environment:
+      # Define rules to scan with by setting the SEMGREP_RULES environment variable. 
+      SEMGREP_RULES: p/default
+      # Scan changed files in PRs or MRs (diff-aware scanning):
+      SEMGREP_BASELINE_REF: << parameters.default_branch >>
+      # Uncomment SEMGREP_TIMEOUT to set this job's timeout (in seconds):
+      # Default timeout is 1800 seconds (30 minutes).
+      # Set to 0 to disable the timeout.
+      # SEMGREP_TIMEOUT: 300
+
+    docker:
+      - image: returntocorp/semgrep
+    steps:
+      - checkout
+      - run:
+          name: "Semgrep scan"
+          command: semgrep ci
+workflows:
+  main:
+    jobs:
+      - semgrep-scan
+```
+
+### Buildkite
+
+To add Semgrep into your Buildkite pipeline:
+
+1. Create or edit a `pipelines.yml` configuration file to add a Semgrep command as part of your pipeline. Refer to the [BuildKite code snippet](#buildkite-code-snippet). This configuration file can also be stored within Buildkite.
+2. Optional: Review the [Buildkite steps definition](https://buildkite.com/docs/pipelines/defining-steps) to help with editing the configuration file.
+3. If you are using Buildkite to store the configuration, save the file. Otherwise, commit the configuration file into the `/.buildkite` folder within the target repository.
+4. The Semgrep job starts automatically upon detecting the `pipelines.yml` commit in the previous step.
+5. Customize the job's behavior, such as its rules to scan and files to ignore.
+
+:::note
+These steps can be performed from within Buildkite's interface. From Buildkite's main page, click **Pipelines > ➕ button** to perform these steps within Buildkite's UI.
+:::
+
+#### Buildkite code snippet
+
+```yaml
+- label: ":semgrep: Semgrep"
+  commands:
+    # Define rules to scan with by setting the SEMGREP_RULES environment variable. 
+    - export SEMGREP_RULES="p/default"
+    # To scan changed files in PRs or MRs (diff-aware scanning):
+    - export SEMGREP_BASELINE_REF=${BUILDKITE_BRANCH}
+    # Uncomment SEMGREP_TIMEOUT to set this job's timeout (in seconds):
+    # Default timeout is 1800 seconds (30 minutes).
+    # Set to 0 to disable the timeout.
+    # - export SEMGREP_TIMEOUT="300"
+    - semgrep ci 
+  
+  plugins:
+    - docker#v3.7.0:
+      image: returntocorp/semgrep
+```
+
+### Running Semgrep through other CI providers
+
+Use either of the following methods to run Semgrep through other CI providers.
+
+#### Direct docker usage 
+
+Reference or add the [returntocorp/semgrep](https://hub.docker.com/r/returntocorp/semgrep) Docker image directly. The method to add the Docker image varies based on the CI provider.
+
+#### Install `semgrep`
```

Comment:
@s-santillan maybe we can change this header to be a bit more specific, so folks will not find it when searching for general "how to install Semgrep".

Reply:
Done in latest push.